### PR TITLE
feat(route): add QuestN

### LINKED ISSUE #12887: RSS Support Request for QuestN
### Category

Social Media

### Website URL

https://app.questn.com/

### Website description

I am submitting this issue to request support for generating RSS feeds from QuestN, a platform that facilitates Web3 campaigns. QuestN allows project owners to create engaging challenges and campaigns, providing users with valuable insights and participation opportunities in various projects.

## Endpoints

The requested support should include three endpoints for generating the desired RSS feeds:

1. **Overall Campaigns**:
   - Endpoint: `https://api.questn.com/consumer/explore/list/`
   - Description: This endpoint will provide a list of overall campaigns hosted on QuestN, allowing users to receive updates on all active campaigns.

2. **Campaigns Filtered by Chain ID**:
   - Endpoint: `https://api.questn.com/consumer/explore/list/?chain_filter=CHAIN_ID`
   - Description: Users can specify a chain ID (e.g., 137) as the parameter to filter campaigns specific to that chain. This endpoint will ensure that RSS feeds contain only campaigns related to the specified chain.

3. **Campaigns Filtered by Community**:
   - Endpoint: `https://api.questn.com/consumer/explore/entity_list/?community_url=COMMUNITY_URL`
   - Description: With this endpoint, users can filter campaigns based on a specific community URL (e.g., Rubic). The resulting RSS feed will include campaigns associated with the specified community.

## RSS Content

The content of the RSS feed for each campaign will include the following details:

- Campaign ID: ID of the campaign on QuestN.
- Title: Title of the campaign.
- Start Time: Timestamp of the campaign start time.
- End Time: Timestamp of the campaign end time.

### Additional description

_No response_

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by cxheng315**
1. Add Route: community, events
2. Add utility file to parse filter string
3. Add QuestN namespace

## Involved Issue / 该 PR 相关 Issue

Close #12887 

## Example for the Proposed Route(s) / 路由地址示例

```routes
/questn/events
/questn/events/chain_filter=1
/questn/events/chain_filter=324&reward_filter=200&status_filter=400
/questn/community/questn
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/questn/community.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { Route } from '@/types';
+import { parseDate } from '@/utils/parse-date';
+import ofetch from '@/utils/ofetch';
+
+export const route: Route = {
+    path: '/community/:community_url',
+    name: 'QuestN Community Events',
```
Comment (TonyRL) [inline lib/routes/questn/community.ts:7]:
```suggestion
    name: 'Community Events',
```

File: lib/routes/questn/community.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    handler,
+};
+
+async function handler(ctx) {
+    const url = 'https://api.questn.com/consumer/explore/entity_list/';
+
+    const params = {
+        count: ctx.req.query('limit') || '20',
+        page: '1',
+        community_url: ctx.req.param('community_url') || 'questn', // default to questn
```
Comment (TonyRL) [inline lib/routes/questn/community.ts:38]: [`camelCase`](https://docs.rsshub.app/joinus/advanced/script-standard#naming)

File: lib/routes/questn/events.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { Route } from '@/types';
+import { parseDate } from '@/utils/parse-date';
+import ofetch from '@/utils/ofetch';
+
+import { parseFilterStr } from './util';
+
+export const route: Route = {
+    path: '/events/:filter?',
+    name: 'QuestN Events',
```
Comment (TonyRL) [inline lib/routes/questn/events.ts:9]:
```suggestion
    name: 'Events',
```

File: lib/routes/questn/events.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    parameters: {
+        category: 'Category of the event', // 100: trending, 200: newest, 300: top
+        status_filter: 'Status of the event', // 0: all, 100: available, 400: missed
+        community_filter: 'Status of the community', // 0: all community, 100: verified, 200: followed
+        rewards_filter: 'Type of the reward', // 0: all rewards, 100: nft, 200: token, 400: whitelist
+        chain_filter: 'Chain id of the event', // 0: all chains, 1: ethereum, 56: bsc, 137: polygon, 42161: arb, 10: op, 324: zksync, 43114: avax
+        search: 'Search keyword',
+        count: 'Number of events to fetch',
+        page: 'Page number',
+    },
```
Comment (TonyRL) [inline lib/routes/questn/events.ts:22]: This should include the description of the path parameter `filter` only
https://github.com/DIYgod/RSSHub/blob/4fe22fb24236cecc9838bbb27cb8bfe404d108f3/lib/routes/questn/events.ts#L8

File: lib/routes/questn/community.ts (hunk 2 above)
Comment (cxheng315) [reply lib/routes/questn/community.ts:38]: this is to fit target api parameter
`https://api.questn.com/consumer/explore/entity_list/?count=20&page=1&community_url=gmnetwork`
so line 41 `${url}?${new URLSearchParams(params)}` would get the correct parameter

Comment (TonyRL) [reply lib/routes/questn/community.ts:38]: I'm talking about the path parameter
https://github.com/DIYgod/RSSHub/blob/2d9cdff0fb5029dbd4c758161fb9f6d243f78ef2/lib/routes/questn/community.ts#L6

Comment (cxheng315) [reply lib/routes/questn/community.ts]: changed, thank you!

**Pull request merged**